core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java

```diff
@@ -55,54 +81,107 @@
  */
 public class DefaultBlockWorkerTest {
-  private TieredBlockStore mTieredBlockStore;
+
+  TieredBlockStore mTieredBlockStore;
+  // worker configurations
+  private static final long WORKER_ID = 30L;
+  private static final long UFS_MOUNT_ID = 1L;
+  private static final WorkerNetAddress WORKER_ADDRESS =
+      new WorkerNetAddress().setHost("localhost").setRpcPort(20001);
+
+  // invalid initial worker id
+  private static final long INVALID_WORKER_ID = -1L;
+
+  // test subject
   private DefaultBlockWorker mBlockWorker;
+
+  // mocked dependencies of DefaultBlockWorker
+  private BlockMasterClient mBlockMasterClient;
   private FileSystemMasterClient mFileSystemMasterClient;
-  private Random mRandom;
-  private String mMemDir =
-      AlluxioTestDirectory.createTemporaryDirectory(Constants.MEDIUM_MEM).getAbsolutePath();
-  private String mHddDir =
-      AlluxioTestDirectory.createTemporaryDirectory(Constants.MEDIUM_HDD).getAbsolutePath();
+
+  // random ID generator
+  private final Random mRandom = new Random();
 
   @Rule
   public TemporaryFolder mTestFolder = new TemporaryFolder();
+  // worker's local storage directories
+  private String mMemDir;
+  private String mHddDir;
+  // ufs file
+  private File mTestUfsFile;
+
   @Rule
   public ConfigurationRule mConfigurationRule =
       new ConfigurationRule(new ImmutableMap.Builder<PropertyKey, Object>()
           .put(PropertyKey.WORKER_TIERED_STORE_LEVELS, 2)
           .put(PropertyKey.WORKER_TIERED_STORE_LEVEL0_ALIAS, Constants.MEDIUM_MEM)
           .put(PropertyKey.WORKER_TIERED_STORE_LEVEL0_DIRS_MEDIUMTYPE, Constants.MEDIUM_MEM)
           .put(PropertyKey.WORKER_TIERED_STORE_LEVEL0_DIRS_QUOTA, "1GB")
-          .put(PropertyKey.WORKER_TIERED_STORE_LEVEL0_DIRS_PATH, mMemDir)
           .put(PropertyKey.WORKER_TIERED_STORE_LEVEL1_ALIAS, Constants.MEDIUM_HDD)
           .put(PropertyKey.WORKER_TIERED_STORE_LEVEL1_DIRS_MEDIUMTYPE, Constants.MEDIUM_HDD)
           .put(PropertyKey.WORKER_TIERED_STORE_LEVEL1_DIRS_QUOTA, "2GB")
-          .put(PropertyKey.WORKER_TIERED_STORE_LEVEL1_DIRS_PATH, mHddDir)
           .put(PropertyKey.WORKER_RPC_PORT, 0)
           .put(PropertyKey.WORKER_MANAGEMENT_TIER_ALIGN_RESERVED_BYTES, "0")
+          .put(PropertyKey.WORKER_BLOCK_HEARTBEAT_INTERVAL_MS, "10ms")
           .build(), Configuration.modifiableGlobal());
   private BlockStore mBlockStore;
 
   /**
    * Sets up all dependencies before a test runs.
    */
   @Before
-  public void before() throws IOException {
-    mRandom = new Random();
-    BlockMasterClient blockMasterClient = mock(BlockMasterClient.class);
+  public void before() throws Exception {
+    // set up storage directories
+    mMemDir = mTestFolder.newFolder().getAbsolutePath();
+    mHddDir = mTestFolder.newFolder().getAbsolutePath();
+    mConfigurationRule.set(PropertyKey.WORKER_TIERED_STORE_LEVEL0_DIRS_PATH, mMemDir);
+    mConfigurationRule.set(PropertyKey.WORKER_TIERED_STORE_LEVEL1_DIRS_PATH, mHddDir);
```

Comment:
I switched from `AlluxioTestDirectory#createTemporaryDirectory` to using `TemporaryFolder`. The reason is better test isolation: `TemporaryFolder` cleans up all test files after test is finished, whereas `AlluxioTestDirectory` relies on a jvm shutdown hook to clean up files after the whole process finished.